modules/cnf-running-cnf-tests-in-discovery-mode.adoc

```diff
@@ -0,0 +1,274 @@
+Existing environment configuration are used for the tests. The tests attempt to find the configuration items needed
+and use those items to execute the tests.
+If resources needed to run a specific test are not found, the test is skipped (providing an appropriate
+message to the user).
+After the tests are finished, no cleanup of the preconfigured configuration items is done, and the test environment
+can immediately be used for another test run.
+
+Some configuration items are still created by the tests.
+These are specific items needed for a test to run; for example, a SRIOV Network.
+These configuration items are created in custom namespaces and are cleaned up after the tests are executed.
+
+An additional bonus is a reduction in test run times.
+As the configuration items are already there, no time is needed for environment configuration and stabilization.
+
+== Prerequisites
+
+Some of the tests require a pre-existing MachineConfigPool to append their changes to.
+This needs to be created on the cluster before running the tests.
+
+The default worker pool is `worker-cnf` and can be created with the following manifest:
+
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: worker-cnf
+  labels:
+    machineconfiguration.openshift.io/role: worker-cnf
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {
+          key: machineconfiguration.openshift.io/role,
+          operator: In,
+          values: [worker-cnf, worker],
+        }
+  paused: false
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker-cnf: ""
+----
+
+You can use the `ROLE_WORKER_CNF` variable to override the worker pool name:
+
+----
+docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig -e
+ROLE_WORKER_CNF=custom-worker-pool quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh
+----
+
+== Enabling discovery mode
+
+To enable discovery mode, the tests must be instructed by setting the `DISCOVERY_MODE` environment variable as follows:
+
+----
+docker run -v $(pwd)/:/kubeconfig:Z -e KUBECONFIG=/kubeconfig/kubeconfig -e
+DISCOVERY_MODE=true quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh
+----
+
+
+== Running the tests
+
+The test entrypoint is `/usr/bin/test-run.sh`. It runs both a setup test set and the real conformance test suite.
+The bare minimum requirement is to provide it a kubeconfig file and it's related `$KUBECONFIG` environment variable
+mounted through a volume.
+
+Assuming the kubeconfig file is in the current folder, the command for running the test suite is:
+
+----
+docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig quay.io/openshift-kni/cnf-tests /usr/bin/test-run.sh
+----
+
+== Image parameters
+The tests can use a different image in the test.
+There are two images used by the tests that can be changed using the following environment variables:
+
+----
+# CNF_TESTS_IMAGE
```

Comment:
I think you can just list these variables out of the code block and apply backticks